src/librustc/mir/mod.rs

```diff
@@ -155,6 +155,8 @@ pub struct Body<'tcx> {
     /// A span representing this MIR, for error reporting.
     pub span: Span,
 
+    pub ignore_interior_mut_in_const_validation: bool,
```

Comment:
Some documentation on this would be great

Reply:
/// This flag is to avoid triggering UnsafeCells are not allowed behind references in constants
    /// for promoteds.

Reply:
/// This flag is to avoid triggering UnsafeCells are not allowed behind references in constants
    /// for promoteds.
    /// It's a copy of `mir::Body`'s ignore_interior_mut_in_const_validation field

Reply:
Copy pasta from the zulip thread:

> The user may be writing e.g. `&[(SOME_CELL, 42)][i].1` and this would get promoted, because we'd statically know that no thing with interior mutability will ever be available to the user without some serious unsafe code.
Now this means that our promoted is actually `&[(SOME_CELL, 42)]` and the MIR using it will do the `&promoted[i].1` projection because the index may be a runtime value. Such a promoted value is illegal because it has reachable interior mutability. This flag just makes this situation very obvious where the previous implementation without the flag hid this situation silently.
> FIXME(oli-obk): rewrite the promoted during promotion to eliminate the cell components.